core/domain/html_validation_service.py

```diff
@@ -749,3 +753,51 @@ def _validate_customization_args_in_tag(tag):
                             yield err_msg
     except Exception as e:
         yield str(e)
+
+
+def add_dimensions_to_noninteractive_image_tag(exp_id, html_string):
+    """Adds dimensions to all oppia-noninteractive-image tags.
+
+    Args:
+        html_string: str. HTML string in which the dimensions is to be
+            added.
+        exp_id: str. Exploration id.
+
+    Returns:
+        str. Updated HTML string with the dimensions for all
+            oppia-noninteractive-image tags.
+    """
+    soup = bs4.BeautifulSoup(html_string.encode('utf-8'), 'html.parser')
+    for image in soup.findAll(name='oppia-noninteractive-image'):
+        filename = unescape_html(image['filepath-with-value'])
+        filename = filename[1:-1]
+        image['filepath-with-value'] = escape_html(json.dumps(
+            get_filename_with_dimensions(filename, exp_id)))
+    return unicode(soup)
+
+
+def get_filename_with_dimensions(filename, exp_id):
+    """Gets the dimensions of the image file.
+
+    Args:
+        filename: str. Name of the file whose dimensions need to be
```

Comment:
Also wrong. You'll want to refer to original_filename and new_filename

Reply:
Done